magento2-nginx/fpm-7.0/usr/local/share/magento2/install_magento.sh

```diff
@@ -22,9 +28,16 @@ fi
 
 if [ ! -d "vendor" ] || [ ! -f "vendor/autoload.php" ]; then
   as_code_owner "composer config repositories.magento composer https://repo.magento.com/"
-  as_code_owner "composer global config http-basic.repo.magento.com '$MAGENTO_USERNAME' '$MAGENTO_PASSWORD'"
-  as_code_owner "composer global config http-basic.toran.inviqa.com '$TORAN_USERNAME' '$TORAN_PASSWORD'"
-  as_code_owner "composer global config github-oauth.github.com '$GITHUB_TOKEN'"
+
+  if [ -n "$MAGENTO_USERNAME" ] && [ -n "$MAGENTO_PASSWORD" ]; then
+    as_code_owner "composer global config http-basic.repo.magento.com '$MAGENTO_USERNAME' '$MAGENTO_PASSWORD'"
+  fi
+  if [ -n "$GITHUB_TOKEN" ]; then
+    as_code_owner "composer global config github-oauth.github.com '$GITHUB_TOKEN'"
+  fi
+  if [ -n "$COMPOSER_CUSTOM_CONFIG_COMMAND" ]; then
+    as_code_owner "$COMPOSER_CUSTOM_CONFIG_COMMAND"
```

Comment:
Ideally it would be in the run_composer method that's in /usr/local/share/php/common_functions.sh - it's already a customisation to not run `--optimize-autoloader`. So yeah, would love to get this done properly!